src/app/modules/color-scheme/color-scheme-contents/color-scheme-contents.component.html

```diff
@@ -0,0 +1,65 @@
+      <div class="square" (click)="colorChanged([1, 4])" [class.currentcolor]="colorSelectedStatus[1][4]"></div>
+      <div class="square" (click)="colorChanged([1, 5])" [class.currentcolor]="colorSelectedStatus[1][5]"></div>
+      <div class="square" (click)="colorChanged([1, 6])" [class.currentcolor]="colorSelectedStatus[1][6]"></div>
+    </div>
+    <div class="swatch three" (click)="schemeChanged(2)">
+      <div class="square" (click)="colorChanged([2, 0])" [class.currentcolor]="colorSelectedStatus[2][0]"></div>
+      <div class="square" (click)="colorChanged([2, 1])" [class.currentcolor]="colorSelectedStatus[2][1]"></div>
+      <div class="square" (click)="colorChanged([2, 2])" [class.currentcolor]="colorSelectedStatus[2][2]"></div>
+      <div class="square" (click)="colorChanged([2, 3])" [class.currentcolor]="colorSelectedStatus[2][3]"></div>
+      <div class="square" (click)="colorChanged([2, 4])" [class.currentcolor]="colorSelectedStatus[2][4]"></div>
+      <div class="square" (click)="colorChanged([2, 5])" [class.currentcolor]="colorSelectedStatus[2][5]"></div>
+      <div class="square" (click)="colorChanged([2, 6])" [class.currentcolor]="colorSelectedStatus[2][6]"></div>
+    </div>
+    <div class="swatch four" (click)="schemeChanged(3)">
+      <div class="square" (click)="colorChanged([3, 0])" [class.currentcolor]="colorSelectedStatus[3][0]"></div>
+      <div class="square" (click)="colorChanged([3, 1])" [class.currentcolor]="colorSelectedStatus[3][1]"></div>
+      <div class="square" (click)="colorChanged([3, 2])" [class.currentcolor]="colorSelectedStatus[3][2]"></div>
+      <div class="square" (click)="colorChanged([3, 3])" [class.currentcolor]="colorSelectedStatus[3][3]"></div>
+      <div class="square" (click)="colorChanged([3, 4])" [class.currentcolor]="colorSelectedStatus[3][4]"></div>
+      <div class="square" (click)="colorChanged([3, 5])" [class.currentcolor]="colorSelectedStatus[3][5]"></div>
+      <div class="square" (click)="colorChanged([3, 6])" [class.currentcolor]="colorSelectedStatus[3][6]"></div>
+    </div>
+    <div class="swatch five" (click)="schemeChanged(4)">
+      <div class="square" (click)="colorChanged([4, 0])" [class.currentcolor]="colorSelectedStatus[4][0]"></div>
+      <div class="square" (click)="colorChanged([4, 1])" [class.currentcolor]="colorSelectedStatus[4][1]"></div>
+      <div class="square" (click)="colorChanged([4, 2])" [class.currentcolor]="colorSelectedStatus[4][2]"></div>
+      <div class="square" (click)="colorChanged([4, 3])" [class.currentcolor]="colorSelectedStatus[4][3]"></div>
+      <div class="square" (click)="colorChanged([4, 4])" [class.currentcolor]="colorSelectedStatus[4][4]"></div>
+      <div class="square" (click)="colorChanged([4, 5])" [class.currentcolor]="colorSelectedStatus[4][5]"></div>
+      <div class="square" (click)="colorChanged([4, 6])" [class.currentcolor]="colorSelectedStatus[4][6]"></div>
+    </div>
```

Comment:
Seems... excessive. Can you give some insight on why some kind of ngFor or otherwise wouldn't work for this?

Reply:
That's just how I had it at the beginning and I never went back to refactor it. I can probably fix it to be less redundant though.